src/test/javascript/spec/component/course/course-exercises.component.spec.ts

```diff
@@ -197,31 +189,31 @@ describe('CourseExercisesComponent', () => {
         ];
 
         // Number of exercises in the course must be 4, since we have added 4 exercises
-        expect(component.course!.exercises!.length).to.equal(4);
+        expect(component.course!.exercises!.length).toEqual(4);
         component.toggleFilters([ExerciseFilter.UNRELEASED]);
 
         // Number of active modeling and text exercises must be 1 respectively, since we have filtered
         // the exercises with release date in the future
-        expect(component.exerciseCountMap.get('modeling')).to.equal(1);
-        expect(component.exerciseCountMap.get('text')).to.equal(1);
+        expect(component.exerciseCountMap.get('modeling')).toEqual(1);
+        expect(component.exerciseCountMap.get('text')).toEqual(1);
 
         component.toggleFilters([ExerciseFilter.UNRELEASED]);
 
         // Number of active modeling and text exercises must be 2 respectively, since we do not filter
         // the exercises with release date in the future anymore
-        expect(component.exerciseCountMap.get('modeling')).to.equal(2);
-        expect(component.exerciseCountMap.get('text')).to.equal(2);
+        expect(component.exerciseCountMap.get('modeling')).toEqual(2);
+        expect(component.exerciseCountMap.get('text')).toEqual(2);
     });
 
     it('should filter all exercises in different situations', () => {
         component.sortingOrder = ExerciseSortingOrder.DESC;
         const filters: ExerciseFilter[] = [ExerciseFilter.OVERDUE, ExerciseFilter.NEEDS_WORK];
-        const localStorageSpy = sinon.spy(localStorageService, 'store');
+        const localStorageSpy = jest.spyOn(localStorageService, 'store');
 
         component.toggleFilters(filters);
 
-        expect(localStorageSpy).to.have.been.calledOnce;
-        expect(component.activeFilters).to.deep.equal(new Set());
+        expect(localStorageSpy).toHaveBeenCalledOnce();
```

Comment:
[nit] According to our coding guidelines, use `toHaveBeenCalledTimes(1)`
```suggestion
        expect(localStorageSpy).toHaveBeenCalledTimes(1);
```